metrics.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2019 RedHat
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metrics
+
+import (
+	"net/http"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// StartMetrics is a function that starts the server with registered metrics on the specified endpoint.
+func StartMetrics(prom prometheusInstance) {
+	// Register Metrics only when the metric List is provided by the operator
+	if prom.metriclist != nil {
+		RegisterMetrics(prom.metriclist)
+		//processMetric()
```

Comment:
What will this be used for?

Reply:
If the user does not provide a list of metrics to register, then the two functions (RegisterMetrics and recordMetricsFunction) are redundant and need not be executed. We can directly expose the generic Go metrics.

Reply:
Sorry! That was the code I commented out. I will remove that line.